internal/cmd/profile/usage_view_csv_command.go

```diff
@@ -0,0 +1,108 @@
+package profile
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/spacelift-io/spacectl/internal/cmd"
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+)
+
+type queryArgs struct {
+	since   string
+	until   string
+	aspect  string
+	groupBy string
+}
+
+func usageViewCSVCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "usage-view-csv",
+		Usage:     "Prints CSV with usage data for the current account",
+		ArgsUsage: cmd.EmptyArgsUsage,
+		Flags: []cli.Flag{
+			flagUsageViewCSVSince,
+			flagUsageViewCSVUntil,
+			flagUsageViewCSVAspect,
+			flagUsageViewCSVGroupBy,
+			flagUsageViewCSVFile,
+		},
+		Before: authenticated.Ensure,
+		Action: func(ctx *cli.Context) error {
+			// prep http query
+			args := &queryArgs{
+				since:   ctx.String(flagUsageViewCSVSince.Name),
+				until:   ctx.String(flagUsageViewCSVUntil.Name),
+				aspect:  ctx.String(flagUsageViewCSVAspect.Name),
+				groupBy: ctx.String(flagUsageViewCSVGroupBy.Name),
+			}
+			params := buildQueryParams(args)
+			req, err := http.NewRequestWithContext(ctx.Context, http.MethodGet, "/usageanalytics/csv", nil)
+			if err != nil {
+				return fmt.Errorf("failed to create an HTTP request: %w", err)
+			}
+			q := req.URL.Query()
+			for k, v := range params {
+				q.Add(k, v)
+			}
+			req.URL.RawQuery = q.Encode()
+
+			// execute http query
+			log.Println("Querying Spacelift for usage data...")
+			resp, err := authenticated.Client.Do(req)
+			if err != nil {
+				return err
+			}
+			defer resp.Body.Close()
+			if resp.StatusCode != http.StatusOK {
+				return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+			}
+
+			// save response to a file
+			var filePath string
+			if !ctx.IsSet(flagUsageViewCSVFile.Name) {
```

Comment:
If this is not set, we could just print to stdout. My reasoning is one could just call it like `spacectl profile usage-view-csv > myfile.csv`, it would also be possible to do additional modifications on the fly if this was possible.

One thing that makes this not great is that we also print things like `log.Println("Querying Spacelift for usage data...")
`

Reply:
I guess a global logging flag would be helpful here. Since we don't have that, what we can do is write csv file to `stdout` and print messages to `stderr`. Although that would be inconsistent with how it's done in other parts of the project. Let me know what you think.

Reply:
> I guess a global logging flag would be helpful here.

I guess, we should maybe create an issue for this.

> Let me know what you think.

I had the idea myself, but didnt suggest it because it is inconsistent. Honestly I have no strong feeling for this. You can merge the code as is.